### PR TITLE
[MM-54803] Convert ./components/channel_header_dropdown/mobile_channel_header_dropdown_animation.tsx from Class to Functional component

### PR DESCRIPTION
**Pull request by Kritik-J**
#### Summary

This PR Convert ./components/channel_header_dropdown/mobile_channel_header_dropdown_animation.tsx from Class to Functional component

#### Ticket Link

fixes: #24766 

#### Jira Ticket

https://mattermost.atlassian.net/browse/MM-54803

### Release Note

```release-note
Convert ./components/channel_header_dropdown/mobile_channel_header_dropdown_animation.tsx from Class to Functional component
```

### CODE REVIEW

File: webapp/channels/src/components/channel_header_dropdown/mobile_channel_header_dropdown_animation.tsx [hunk 1]
```diff
@@ -10,26 +10,25 @@ const ANIMATION_DURATION = 350;
...
+            in={show}
+            classNames='mobile-channel-header-dropdown'
+            enter={true}
+            exit={true}
+            mountOnEnter={true}
+            unmountOnExit={true}
+            timeout={{
+                enter: ANIMATION_DURATION,
+                exit: ANIMATION_DURATION,
+            }}
```
Comment (larkox) [inline webapp/channels/src/components/channel_header_dropdown/mobile_channel_header_dropdown_animation.tsx:27]: Can we define this outside of the component? If not, on every render, we are passing a new value to `timeout`, potentially triggering a re-render underneath.

**Review by larkox — CHANGES_REQUESTED** [review]
Great work! Just a minor change.

**Review by larkox — APPROVED** [review]

**Review by jespino — APPROVED** [review]
Nice! Looks good to me. Thanks @Kritik-J! :tada:

**Comment by josephjosedev** [comment]
@lindy65  i would like to test this PR as a QA

**Comment by lindy65** [comment]
Thanks @josephjosedev - the test server is ready.  You will need to test this on the [latest build of the mobile app](https://mattermost.com/download/#mobile) as the channel drop-down menu needs to be smoketested.

Let me know if you have any questions :)

**Comment by lindy65** [comment]
Hi @josephjosedev - yes - tap on the name "Off Topic" and test some of the options that you see in the menu that appears after tapping on the channel name.

Also, create your own new public and private channels and test the channel drop-down from these newly created channels

**Comment by larkox** [comment]
@lindy65 @josephjosedev The idea here is not so much about the mobile application, but mobile web browser. You can test this either by opening the mattermost server from a browser in the phone (instead of from the app), or reduce the width of the browser to a really narrow size. Sorry for the confusion!

**Comment by lindy65** [comment]
> @lindy65 @josephjosedev The idea here is not so much about the mobile application, but mobile web browser. You can test this either by opening the mattermost server from a browser in the phone (instead of from the app), or reduce the width of the browser to a really narrow size. Sorry for the confusion!

Aah!  Thanks for the clarification @larkox :)

**Comment by josephjosedev** [comment]
i tested the  options on the channel dropdown and some bugs are noted and the noted issues are attaching below
image
1.On channel Action Option the placeholder text is cutoffed..
2. 
image
Convert to private channel option 2 buttons are touching Yes and No button

3.
image
Add members invite to the team text is congested in the space..

**Review by lindy65 — APPROVED** [review]
Thanks @Kritik-J  - looks good 👍 

@josephjosedev - thanks for your thorough QA!  I'll file separate tickets for the issues you found as they are not caused by this PR

cc @larkox

**Comment by larkox** [comment]
Thank you for the contribution!

**Pull request merged**